packages/formation/js/additional-info.js

```diff
@@ -43,22 +49,28 @@ export default function createAdditionalInfoWidget() {
       const chevron = qS(el, 'i.fa-angle-down');
       const button = qS(el, 'button');
 
-      const additionalInfoClickEvent = document.createEvent('Event');
-      additionalInfoClickEvent.initEvent(
-        '@department-of-veterans-affairs/formation/additional-info/button-clicked',
-        true,
-        true,
-      );
-      additionalInfoClickEvent.detail = { titleText, dataset };
+      const additionalInfoExpandEvent = document.createEvent('Event');
+      additionalInfoExpandEvent.initEvent(EVENT_NAMES.EXPAND, true, true);
+
+      const additionalInfoCollapseEvent = document.createEvent('Event');
+      additionalInfoCollapseEvent.initEvent(EVENT_NAMES.COLLAPSE, true, true);
 
       button.addEventListener('click', () => {
         const ariaExpanded = JSON.parse(button.getAttribute('aria-expanded'));
+        const isExpanded = !ariaExpanded;
 
-        button.setAttribute('aria-expanded', `${!ariaExpanded}`);
-        button.parentNode.classList.toggle('form-expanding-group-open');
-        chevron.classList.toggle('open');
+        button.setAttribute('aria-expanded', `${isExpanded}`);
+        button.parentNode.classList.toggle(
+          'form-expanding-group-open',
+          isExpanded,
+        );
+        chevron.classList.toggle('open', isExpanded);
 
-        button.dispatchEvent(additionalInfoClickEvent);
+        if (isExpanded) {
+          el.dispatchEvent(additionalInfoExpandEvent);
+        } else {
+          el.dispatchEvent(additionalInfoCollapseEvent);
+        }
```

Comment:
Really open to discussion here. Is separating the event discouraged? Should the button be the trigger, instead of the container, since that's where the event originated?

Reply:
Exactly like you said, I would do `button.dispatchEvent(additionalInfoExpandEvent);`since that's where the event is propagating from :) Separating the event with the conditional makes sense to me though!